### PR TITLE
CDAP-9125 CDAP-11929 Misc authorization fixes

### PR DESCRIPTION
**Pull request by poornachandra**
Fixes for:
https://issues.cask.co/browse/CDAP-9125
https://issues.cask.co/browse/CDAP-11929

### CODE REVIEW

File: cdap-sentry/cdap-sentry-extension/cdap-sentry-binding/src/main/java/co/cask/cdap/security/authorization/sentry/binding/AuthBinding.java [hunk 1]
```diff
@@ -425,6 +425,15 @@ public Void run(SentryGenericServiceClient client) throws Exception {
         if (principal.getType().equals(Principal.PrincipalType.USER)) {
           // for a user get all the groups and their roles
           Set<String> groups = authProvider.getGroupMapping().getGroups(principal.getName());
+          LOG.debug("Got groups {} for principal {}", groups, principal);
+          if (!groups.contains(principal.getName())) {
+            LOG.debug("Principal {} does not belong to group {}. Adding the principal's group {} to the groups for " +
+                        "enforcement",
```
Comment (poornachandra) [inline cdap-sentry/cdap-sentry-extension/cdap-sentry-binding/src/main/java/co/cask/cdap/security/authorization/sentry/binding/AuthBinding.java]: If cannot avoid line break for the debug message, wrap it in `isDebugEnabled()` condition.

**Review by poornachandra — COMMENTED** [review]
@rohitsinha54 Just one comment, please fix and merge.

**Comment by poornachandra** [comment]
LGTM

**Comment by yaojiefeng** [comment]
LGTM

**Pull request merged**